### ISSUE TITLE
Build fails in Xcode 9 beta 3

### ISSUE THREAD

**Issue by Babelfisch** [Skyscanner/SkyFloatingLabelTextField#158]
It's not possible to build the project on Xcode beta 3:

![screenshot 2017-07-21 08 26 56](https://user-images.githubusercontent.com/1218405/28451619-8f3649c4-6dee-11e7-984a-8ae208ee6803.png)

Changing the Swift Language Settings to 3.2 has no effect.

**Comment by Babelfisch** [comment]
The error only occurs if I install SkyFloatingLabelTextField als Cocoa Pod. The test project from GitHub itself works fine in Xcode 9 beta 3.

**Comment by k0nserv** [comment]
This looks like Swift 4 changes right @bogren?

**Comment by bogren** [comment]
Yes, building for 3.2 works fine in Xcode 9. I suggest we create a `swift-4`branch  🙂

**Comment by Babelfisch** [comment]
👍 for swift-4 branch

**Comment by KrLx1994roller** [comment]
Then what do I have to do to be able to use it using CocoaPods?

**Comment by nalexander50** [comment]
Swift 4 user here. Can we get an ETA on the Swift 4 branch? Looked like there were a couple PRs that just haven't been merged yet. Having to unlock and edit the framework manually is making me feel dirty!

**Comment by k0nserv** [comment]
Sorry if my understanding of this is incorrect, but shouldn't Xcode 9 be able to build both Swift 3 and Swift 4 libraries as part of the same target?

**Comment by k0nserv** [comment]
This seems related to this issue in CocoaPods. https://github.com/CocoaPods/CocoaPods/issues/6791

I'm having a look at supporting both Swift 3.2 and Swift 4, but for now the workaround there should resolve the issue of Swift 4. Can you try it?

**Comment by nalexander50** [comment]
I didn't realize this was an on-going, open issue for the Cocoapods team. Looks like the post install hook should work well enough for now -- I will try it when I get home. It's not pretty but it's less ugly than unlocking and manually editing the framework. Hopefully Cocoapods can come up with an elegant solution to this problem for the future. Similarly, it would also be nice if I could get Xcode 9 to stop nagging me about migrating Pod code to Swift 4 syntax -- having a single phantom warning is really irritating.

I agree that you should avoid making a breaking change when it's not necessary; however, there is something to be said about not burdening users who are using the latest versions (not that adding some copy-paste lines to the Podfile is *really* a burden, just an expression). Granted, it would become unwieldy to maintain `N` branches to support `N` versions of Swift as the years go by. I noticed that PromiseKit uses some `#if swift(>=4.0)` checks to help with compatibility but that can really make your codebase look ugly and may not be the best solution in some cases. This is definitely an interesting problem and I'm curious to see how Cocoapods addresses it.

At any rate, thanks for the prompt response. I'll try out that post install hook.

**Comment by k0nserv** [comment]
Yeah the approach I want to look at is something similar to `PromiseKit` using conditional code paths to support both Swift 3 and Swift 4. The time I can dedicate to SkyFloatingLabelTextField is a bit limited unfortunately, but I've started hacking on it

**Comment by nalexander50** [comment]
Looks like it's working just fine.

**Comment by k0nserv** [comment]
I've just pushed a new commit(a5fe9c9) that adds support for both Swift 3.2 and Swift 4 at the same time. Can everyone please test this with

```ruby
pod 'SkyFloatingLabelTextField', git: 'https://github.com/Skyscanner/SkyFloatingLabelTextField/'
```

**Comment by DaRkRaGe** [comment]
It's working now with this pod: SkyFloatingLabelTextField', git: 'https://github.com/Skyscanner/SkyFloatingLabelTextField/'

**Comment by k0nserv** [comment]
Just released 3.2.1 with this fix. Closing the issue now
